http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -205,7 +205,11 @@ public void onOpen(final WebSocket webSocket) {
     public CompletionStage<?> onClose(
         final WebSocket webSocket, final int statusCode, final String reason) {
       pingSender.cancel();
-      listener.connectionClosed(reason);
+      if (reason.equals(WebSocketConnection.CLIENT_DISCONNECT_MESSAGE)) {
+        listener.connectionClosed();
+      } else {
+        listener.connectionTerminated(reason.isBlank() ? "Server disconnected" : reason);
+      }
```

Comment:
I don't know what the server side of the websocket implementation is capable of, but I think it would be a better idea to use a custom status code  for closing (it looks like codes 4000 - 4999 are reserved for application use) and make decisions based on that.

Reply:
Why better?

status code for http calls can certainly give more information. There are cases where the payload of a 200 is an error message, ie: the message and service are all okay, everything happened as expected, but something resulted in an error.

In this case though, the message string is quite exact, keeps it quite simple. (For example, a close from server very well could be a 200, which would look like the same if the client closed)